Calibration/PPSAlCaRecoProducer/test/test_prompt_AlCaRecoProducer.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+customise_commands="process.GlobalTag.toGet = cms.VPSet()\
+\nprocess.GlobalTag.toGet.append(cms.PSet(record = cms.string(\"AlCaRecoTriggerBitsRcd\"),tag =  cms.string(\"AlCaRecoHLTpaths_PPS2022_prompt_v1\"), connect = cms.string(\"frontier://FrontierProd/CMS_CONDITIONS\")))\
+\nprocess.GlobalTag.toGet.append(cms.PSet(record = cms.string(\"PPSTimingCalibrationLUTRcd\"),tag =  cms.string(\"PPSDiamondTimingCalibrationLUT_test\"), connect = cms.string(\"frontier://FrontierProd/CMS_CONDITIONS\")))\
+\nprocess.ALCARECOPPSCalMaxTracksFilter.TriggerResultsTag = cms.InputTag(\"TriggerResults\",\"\",\"HLTX\")"
+
+# note we currently use `auto:run3_data_express` GT 
+# the correct GT (auto:run3_data_prompt) doesn't have LHCInfo record for run 322022 which corresponds to our face ALCARAW file
+cmsDriver.py testPromptPPSAlCaRecoProducer  -s ALCAPRODUCER:PPSCalMaxTracks,ENDJOB \
+--process ALCARECO \
+--scenario pp \
+--era ctpps_2018 \
+--conditions auto:run3_data_express \
+--data  \
+--datatier ALCARECO \
+--eventcontent ALCARECO \
+-n 100  --filein file:/build/lgrzanka/outputALCAPPS_single.root \
+--fileout file:outputALCAPPS_RECO_prompt.root \
+--customise_commands="$customise_commands"
```

Comment:
for the record this could easily become a unit test, to be able to test the workflow even in absence of appropriate input files discoverable from DAS. See https://github.com/mmusich/cmssw/commit/90148986c6c214c871f20a174c246a93b5d33269, feel free to take it (or dismiss it).
Notice that the input file should be copied at `/store/group/alca_global/unit-tests/outputALCAPPS_single.root` (or any other location of your preference)

Reply:
Thanks, good idea! The only concern I have is that the `*_output.py` files (which are faulty BTW - the InputTags are not updated; I'll fix that) are not included in the unit tests. As they are supposed to be run after the corresponding shell files (express/prompt), would it be acceptable to "extract" only one part of the workflow to a unit test?

Reply:
@MatiXOfficial 

> The only concern I have is that the *_output.py files are not included in the unit tests.

That is relatively straightforward to implement e.g. you can take a look to:

https://github.com/mmusich/cmssw/commit/0caf2d0bd3b74e013275d24c14807c1e60abba68

(which makes use of the prerequisite test flag), but there are other ways of doing it (e.g. adding the `cmsRun` command for the output after the cmsDriver for the first step in the other bash scripts).

Reply:
That's straightforward indeed :) I'll add this change tomorrow or after the weekend.

Reply:
@mmusich I tried to follow your approach and create two new tests for the `*_output.py` files (express/prompt) with the `PRE_TEST` flag, but the output tests could not read the files created by the producer tests (`Permission denied`). 

In the end, I just put the `cmsRun` command after `cmsDriver` and now the tests seem to be working. 

The last issue is that I don't have the write permission in `/eos/cms/store/group/alca_global/` so I cannot create a new folder. Do you have the write permission there? If so, could you create a new folder named `pps_alcareco_producer_tests` (I guess this is a proper name considering that another file - https://github.com/cms-sw/cmssw/pull/36273#discussion_r805017506 - might be put there) and copy the `/eos/user/k/kocotm/Shared/alcarecoproducer/outputALCAPPS_single.root` file there? You should have access to that file.

Reply:
@MatiXOfficial I created the directory:
`/eos/cms/store/group/alca_global/pps_alcareco_producer_tests`
but your file is in reading mode only, so I cannot copy it there.

Reply:
I didn't mean the `.sh` files - they are executed when I type `scram b runtests`. The problem is with the `root` files created by the `test_*_AlCaRecoProducer.sh` files. They are needed in the `*_output.py` files. When the `*_output.py` files are executed in the same script as the cmsDriver command, they can read the `root` files without any problems. If I put them in separate scripts, then they cannot read the `root` files anymore.

@tvami In such case, could you create that folder and copy the `outputALCAPPS_single.root` file there? You should be able to access this file as well.

Reply:
@MatiXOfficial Francesco just commented that your file has reading mode only, so we cant copy it.

Reply:
@francescobrivio @tvami Sorry, I forgot about that... Now it should be all right.

Reply:
ok it works! I've copied the root file now:
`/eos/cms/store/group/alca_global/pps_alcareco_producer_tests/outputALCAPPS_single.root`

Reply:
@MatiXOfficial 

>  If I put them in separate scripts, then they cannot read the root files anymore.

This sounds very strange.

Reply:
@mmusich Thanks, but it turns out you were right from the beginning, and I misunderstood the error message... I gave the proper permissions to the new `.sh` files and now they work correctly.